pydantic/color.py

```diff
@@ -45,13 +45,13 @@ def __getitem__(self, item: Any) -> Any:
 # these are not compiled here to avoid import slowdown, they'll be compiled the first time they're used, then cached
 r_hex_short = r'\s*(?:#|0x)?([0-9a-f])([0-9a-f])([0-9a-f])([0-9a-f])?\s*'
 r_hex_long = r'\s*(?:#|0x)?([0-9a-f]{2})([0-9a-f]{2})([0-9a-f]{2})([0-9a-f]{2})?\s*'
-_r_255 = r'(\d{1,3}(?:\.\d+)?)'
+_r_255 = r'((?:\d|\d\d|[01]\d\d|2[0-4]\d|25[0-4])(?:\.\d+)?|255(?:\.0+)?)'
```

Comment:
I think best to add add a separate PR for these changes, sorry.

Reply:
Done - that's #2155.  

Because this PR requires the more precise regex patterns, it's rebased on that PR but they can be merged separately.